### PR TITLE
Fix URL parsing for windows-like paths

### LINKED ISSUE #182: UrlParser cannot distinguish between URLs and Windows paths
UrlParser uses the `URL` constructor to determine whether the string passed is a browser URL or not.

Unfortunately, absolute windows paths (e.g. `C:\an\absolute\path`) are successfully parsed as URLs by `URL`. (e.g. `C:\an\absolute\path` having `protocol: 'c:'` and `pathname: '\\an\\absolute\\path'`)

This results in the windows path being sent to `request` leading to an error of the form `Error: Invalid URI "c:/an/absolute/path"`.

### PR DESCRIPTION
**Pull request by spalladino**
Fixes #182

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -14,7 +14,12 @@
     "lint": "tslint -c ../../tslint.json -p ./"
   },
   "dependencies": {
+    "@types/is-url": "^1.2.28",
```
Comment (ritave) [inline packages/core/package.json]: Move to dev-dependencies

File: packages/core/src/resolvers/uriresolver.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
+import isUrl from "is-url";
 import { Context } from "..";
 import { SubResolver } from "./subresolver";
 
 export function UriResolver(): SubResolver {
   // TODO: make a resolver that actually checks if something is a valid http link
```
Comment (ritave) [inline packages/core/src/resolvers/uriresolver.ts]: I'd say this comment is now deprecated, can be removed to not confuse developers

**Comment by spalladino** [comment]
Sorry @ritave! I kept developing on top of master on our fork, with additional changes in order to release for supporting the latest release of zOS, and forgot that this PR was still pulling those changes. I'll change the head branch now!

**Comment by spalladino** [comment]
Closed in favor of #184

**Pull request closed without merging**